pyiceberg/transforms.py

```diff
@@ -349,6 +356,13 @@ def dedup_name(self) -> str:
     def preserves_order(self) -> bool:
         return True
 
+    @abstractmethod
+    def pyarrow_transform(self, source: IcebergType) -> "Callable[[pa.Array], pa.Array]": ...
```

Comment:
Shall we put this in the super class `Transform`, in the future we will support all the transforms and they will share the same method def for this